### DISCUSSION TITLE
CRUD GET Multiple Response Records

### DISCUSSION THREAD

**Discussion by neilangell** [mockoon/mockoon#1418] [Ideas]
I am trying to use a CRUD route to simulate an API relating to event type data where there will be multiple records for a specific resource id. See simple example JSON data bucket below. I’d like to set the id field for the CRUD route to “resId” so that when I call GET …/1 I get 3 records back. At the moment I only get back 1 record, the first one. Can I make Mockoon return all 3? If not can I create an Issue for this to go into your backlog, as this would be really useful feature I my view? Many thanks.

[
	{ evtId: 101, resId: 1 },
	{ evtId: 103, resId: 1 },
	{ evtId: 105, resId: 1 },
	{ evtId: 102, resId: 2 },
	{ evtId: 104, resId: 2 }
]

**Comment by 255kb** [comment]
We tried to follow REST conventions with the CRUD feature, and what you are describing, using `/res/:id` to filter is not the usual convention. 

[Filtering](https://mockoon.com/docs/latest/api-endpoints/crud-routes/#filtering-sorting-and-pagination-on-the-main-get-route) is usually achieved using query parameters: `/res?resId_eq=1`
